### ISSUE TITLE
Don't display unnecessary warning if no target framework is defined

### ISSUE THREAD

**Issue by dcorriveau-omniscient** [fsprojects/Paket#1455]
Related to changes in fsprojects/Paket/issues/1442
The warning seems to display even when it shouldn't, we might need to be more restrictive about it.
`X contains libraries, but not for the selected TargetFramework net452 in project Y`

For example, I have a solution with projects that target net452, and I don't define any version or framework requirements in the dependencies file. I still get tons of warning for libraries such as `Microsoft.CSharp`, `Microsoft.Net.Http` or `System.Net.Http`.

**Comment by forki** [comment]
But the message is correct in this case, right?

**Comment by dcorriveau-omniscient** [comment]
I don't know what's the right solution for this. But I do know that I don't want tons of warnings in the output window as soon as I add a reference like EntityFramework. At the same time we want the warnings when VS is failing to import dependent assemblies when there is a framework mismatch.

**Comment by forki** [comment]
what happens when you enable http://fsprojects.github.io/Paket/dependencies-file.html#Automatic-framework-detection ?

**Comment by dcorriveau-omniscient** [comment]
When I paket update I get the error
`Could not detect target framework for project x.csproj`
and it stops the process

**Comment by forki** [comment]
ouch. that's clearly a bug ;-)

**Comment by forki** [comment]
please retry

**Comment by matthid** [comment]
I get a warning when `paket update` on FSharp.Formatting
```
Package Microsoft.AspNet.Razor contains libraries, but not for the selected TargetFramework net40 in project C:\PROJ\FSharp.Formatting\tests\FSharp.MetadataFormat.Tests\files\crefLib\crefLib1.fsproj.
```
The point is that the project is not even referencing the package. I think this is related to the fact that paket chooses the `paket.references` from the upper directory. Until now we never cared (besides some additional references nothing happened), but now we get several warnings.

Maybe its time to add an empty `paket.references` file. (This is just to let others know if they stumble over this issue for the same reason)

**Comment by forki** [comment]
I think non-existing paket.references is same as empty file. So the issue is elsewhere

**Comment by matthid** [comment]
I'm pretty sure it's not: Paket searches up the directories until it finds a `paket.references` file and uses that. (If you say this is a bug that would be nice, because I never liked that behavior anyway; but this is a (possibly huge) breaking change). Paket does this for quite some time now.

**Comment by forki** [comment]
Ah you are right. That was a drunk decision :-)

**Comment by dcorriveau-omniscient** [comment]
> please retry
Same thing :(

**Comment by forki** [comment]
Yeah. Damn it. I can't reproduce it.

**Comment by dcorriveau-omniscient** [comment]
Did you try with a reference to `System.Net.FtpClient` ?

**Comment by forki** [comment]
Yes. But it doesn't pick the csproj up. Need to investigate deeper. Or can
you give me a mini repro?
I know this is far away from the original issue ;-)

**Comment by dcorriveau-omniscient** [comment]
I think you need more than my project, it seems to be tied to the fact that I'm using custom packages from my private nuget server that are built using the "project" type with paket, and those are referencing System.Net.FtpClient.

**Comment by davidschwegler** [comment]
This discussion is already a bit of all over the place, but this seems like the best place to discuss it, and I have an issue with it ~~

I get pages of errors like these even though my project is fine, making the output really hard to read and potentially hiding bugs: 
``Package Newtonsoft.Json contains libraries, but not for the selected TargetFramework net45 in project C:\Code\MyProject\src\MyProject.Core\MyProject.Core.csproj.``

My project.dependencies file has something like

``
framework: portable-net45+win, MonoAndroid, XamariniOS
...
nuget Newtonsoft.Json 6.0.8.3
``

In our portable projects, the TargetFrameworkVersion gets set as ``v4.5``, TargetFrameworkProfile ``Profile7``, and TargetFrameworkIdentifier ``.NETPortable``. 

To reduce the ``choose`` elements in our csproj, I'm setting the framework explicitly in the paket.references file to
``Newtonsoft.Json framework: portable-net45+win``

If I understand this Warning's code correctly, it's checking that the csproj has set ``TargetFrameworkVersion`` to ``v4.5``, and the regardless of the fact I'm excluding that framework in paket.references, it's complaining because it's not including it in the project.

Is this a bug, or a misunderstanding on my part?

**Comment by forki** [comment]
@davidschwegler I think that is an edge case we don't cover yet. Can you create a minimal repro with the settings you described an put it into a zip or something?

**Comment by smoothdeveloper** [comment]
BTW, I'm noticing this on transitive dependencies that don't apply for the target framework. I get this via `Zlib.Portable` pulled by `FSharp.Data`.

Will put a integration test if I have a chance.

**Comment by MovGP0** [comment]
a warning when a library for a given target framework is missing is fine. 

the bigger issue is that the warning is not providing enough information to fix the issue. it would be helpful to know which target frameworks are supported by a given nuget package, so those should be listed.
